databases/week1/Week1-homework.sql

```diff
@@ -0,0 +1,91 @@
+/* Find out how many tasks are in the task table */
+select
+  COUNT(id)
+from
+  task;
+
+/* Find out how many tasks in the task table do not have a valid due date */
+select
+  COUNT(id)
+from
+  task
+where
+  IsNull(due_date);
+
+/* Find all the tasks that are marked as done */
+select
+  *
+from
+  task
+where
+  status_id = 3;
+
+/* Find all the tasks that are not marked as done */
+select
+  *
+from
+  task
+where
+  status_id != 3;
+
+/* Get all the tasks, sorted with the most recently created first */
+select
+  *
+from
+  task
+ORDER BY
+  created DESC;
+
+/* Get the single most recently created task */
+select
+  *
+from
+  task
+ORDER BY
+  created DESC
+LIMIT 1;
+
+/* Get the title and due date of all tasks where the title or description contains database */
+select
+  *
+from
+  task
+where
+  title like '%database%'
+  OR
+  description LIKE '%database%';
+
+/* Get the title and status (as text) of all tasks */
+select
+  title,
+  case
+    WHEN status_id = 1 THEN 'Not started'
+    WHEN status_id = 2 THEN 'In progress'
+    WHEN status_id = 3 THEN 'Done'
+    ELSE ''
+  END AS status
+from
+  task;
```

Comment:
Smart here. But take care of sensitivity as we already have the status names that we can retrieve from table status.

Reply:
I can understand.

Thank you for your valuable time. Nice to see the comments. It improves my code.